src/mailingui/types/types.ts

```diff
@@ -0,0 +1,35 @@
+import { CSSProperties } from "react";
+
+export interface Variant {
+  color: CSSProperties["color"];
+  backgroundColor: CSSProperties["backgroundColor"];
+  borderColor?: CSSProperties["borderColor"];
+}
+
+export interface BadgeVariant extends Variant {
+  dot?: boolean;
+}
+
+export interface Theme {
+  fontFamily?: string;
+  borderRadius?: number;
+  buttonVariants?: ButtonVariants;
+  badgesVariants?: BadgeVariants;
+}
+
+type ButtonVariants = {
+  primary: Variant;
+  secondary: Variant;
+  soft: Variant;
+};
+
+type BadgeVariants = {
+  default: BadgeVariant;
+  danger: BadgeVariant;
+  success: BadgeVariant;
+  info: BadgeVariant;
+  warning: BadgeVariant;
+};
+
+export type ButtonVariantKey = keyof ButtonVariants;
+export type BadgeVariantsKey = keyof BadgeVariants;
```

Comment:
- I am very inclined in suggesting these types to be with their components. 
- I also don't like the specific buttonVariants to be in the theme. It's not very scalable, imagine we have 12 components. Would that mean that we would include all of them in the theme?
- Again, none of these feel to me that they really belong in a dedicated type directory or shared types file.

Reply:
- I deleted the variants for each component and included singular variants, as we talked about.
- I did not create new types for variants inside each component, so right now, every component (only `Badge` and `Button`) can use all of the variants defined in the Theme.
- I left the `types` file in the project and only modified it, so please check it out if that's something you agree with.